packages/aws_common/lib/src/io/aws_file_platform_html.dart

```diff
@@ -125,18 +125,25 @@ class AWSFilePlatform extends AWSFile {
           return externalContentType;
         }
 
-        String blobType;
+        String? blobType;
 
         final file = _inputFile ?? _inputBlob;
         if (file != null) {
           blobType = file.type;
-        } else {
+        }
+
+        final path = super.path;
+        if (path != null) {
           blobType = (await _resolvedBlob).type;
```

Comment:
Are you overriding blobType on purpose? Are these branches mutually exclusive? Can you add some notes here?

Reply:
Ugh didn't commit the final change on this file. Will update.

Reply:
Updated the logic resolve `blobType` from `File (Blob)` or from `Blob` resolved from path. The overriding was incorrect.

Reply:
Can you add another test for this since it wasn't caught before?

Reply:
Added 👍